src/common/config.cc

```diff
@@ -883,6 +883,12 @@ int md_config_t::get_val(const std::string &key, char **buf, int len) const
   return _get_val(key, buf,len);
 }
 
+int md_config_t::get_val(const std::string &key, std::string *val) const
```

Comment:
What about giving this a different name to reflect that it's a printer?  Like "get_val_as_string" or somesuch